components/ntp_background_images/browser/view_counter_service.cc

```diff
@@ -108,35 +112,55 @@ void ViewCounterService::BrandedWallpaperWillBeDisplayed(
 }
 
 NTPBackgroundImagesData*
+ViewCounterService::GetCurrentWallpaperData() const {
+  return service_->GetBackgroundImagesData();
+}
+
+NTPSponsoredImagesData*
 ViewCounterService::GetCurrentBrandedWallpaperData() const {
-  auto* sr_data = service_->GetBackgroundImagesData(true /* for_sr */);
+  auto* sr_data = service_->GetBrandedImagesData(true /* for_sr */);
   if (sr_data && IsSuperReferralWallpaperOptedIn())
     return sr_data;
 
-  return service_->GetBackgroundImagesData(false);
+  return service_->GetBrandedImagesData(false);
 }
 
 base::Value ViewCounterService::GetCurrentWallpaperForDisplay() const {
-  if (ShouldShowBrandedWallpaper()) {
+  bool shouldShowBrandedWallpaper = ShouldShowBrandedWallpaper();
+  if (shouldShowBrandedWallpaper) {
+    return GetCurrentBrandedWallpaper();
+  } else {
     return GetCurrentWallpaper();
   }
 
   return base::Value();
 }
 
 base::Value ViewCounterService::GetCurrentWallpaper() const {
+  //BI component has a delay between initialization and downloading finish, need to
```

Comment:
Is this comment still valid? AFAIK, Android NTP will ask wallpaper data when component is ready.